### ISSUE TITLE
Add debug option to show raw SQL

### ISSUE THREAD

**Issue by igorsantos07** [cakephp/phinx#93]
Sometimes when trying to run migration and failing for weird database errors it's hard to debug the problem without knowing what exact SQL is being issued against the server. The `--verbose` option shows what Phinx is trying to do, but not exactly how.

Maybe another option `-vv` or `--debug` could show the commands sent to the server?

**Comment by TrogWarZ** [comment]
+1

**Comment by valkirilov** [comment]
+1

**Comment by rquadling** [comment]
+1

Maybe also report back the number of rows affected/inserted too.

**Comment by jiMuBao** [comment]
+1

**Comment by Keloo** [comment]
+1

**Comment by rquadling** [comment]
I'd be happy with the verbose flag being used if the intent is that you occasionally (i.e. during debugging/building) you want to see the queries.

I've got a PR for showing the migration start/end times. Using ``-v``.

For this issue, a ``-v`` (w|c|sh)ould show the query, with ``-vv`` showing maybe the response times and rows affected counts.

Considering Symfony/Console already provides this option, it would seem pointless creating another for something that is of occasional use.

Having said that, a ``--query-log[=filename]`` option would also make sense. With the filename being defaulted to something like ``phinx_2015-10-22_17-46-17_migrate.sql`` and ``phinx_2015-10-22_17-46-17_rollback.sql`` so you could see a list of SQL files, in order of execution, containing JUST the SQL statements executed.

**Comment by TiGR** [comment]
+1

**Comment by daniel-gomes-sociomantic** [comment]
+1

**Comment by daniel-gomes-sociomantic** [comment]
this seems to be somewhat related to https://github.com/robmorgan/phinx/issues/567 (ie. as both features would show the SQL statements, the one here running them after and that other exiting without running them)

**Comment by chrisforrence** [comment]
+1

**Comment by SanderSander** [comment]
+1

**Comment by mfacenet** [comment]
+1

**Comment by FlorianH** [comment]
+1

**Comment by rewmike** [comment]
+1

**Comment by stromaler** [comment]
+1

**Comment by rquadling** [comment]
A potential issue arises where a migration requires data to operate upon. The initial ``SELECT`` would be reported, but by not actually being executed, the remainder of the migration would not be processed. Potentially, maybe as a further logging option to allow non destructive queries, allowing ``SELECT`` statements through, but not ``UPDATE``, ``DELETE``, ``INSERT``, etc. could be useful.

The quality of the results would obviously depend upon the query and the potential for an infinite loop is present; i.e. Query is for the last non-updated row, loop until all rows are processed. As the ``SELECT`` will always return the first non-updated row, there is no exit.

So, a consideration really. Not a requirement (though it would be nice to have if infinite loop detection existed .... and how for out of scope is that!!! Ha!)

**Comment by fefas** [comment]
+1

This feature would be useful to when working with replication tools like `slony` that in some cases requires`SQL` code used on the master node to sync the new data or tables to the slave node.

**Comment by macwinnie** [comment]
+1

@tomzx provided a simple adaption above, that could easily be implemented.

**Comment by ahukkanen** [comment]
+1

**Comment by JamesTheHacker** [comment]
+1

**Comment by JamesTheHacker** [comment]
@tomzx this doesn't work for me. The only output I get is `START TRANSACTION`. It doesn't print any SQL before it fails.

**Comment by leonkielstra** [comment]
+1

**Comment by lorenzo** [comment]
Already implemented in `v0.10.0` with the `--verbose` flag

**Comment by arnisjuraga** [comment]
Just a note - running `--dry-run` displays raw SQL!


```sql
START TRANSACTION
ALTER TABLE `product` CHANGE `date_start` `date_start` DATE NULL DEFAULT ''
ALTER TABLE `product` CHANGE `date_end` `date_end` DATE NULL DEFAULT ''
update product set date_start = NULL where date_start < '1971-01-01'
update product set date_end = NULL where date_end < '1971-01-01'
ALTER TABLE `product` ADD  KEY (`customer_group_id`,`date_start`,`date_end`)
COMMIT
INSERT INTO migrations (`version`, `migration_name`, `start_time`, `end_time`, `breakpoint`) VALUES ('20190709093643', 'testMigration', '2019-07-12 09:04:56', '2019-07-12 09:04:56', 0);

```
